src/cert/client_cert_store.cpp

```diff
@@ -34,33 +31,68 @@ namespace
 {
 constexpr auto chain_name = "multipass_client_certs.pem";
 
-void validate_certificate(const std::string& pem_cert)
+auto load_certs_from_file(const multipass::Path& cert_dir)
 {
-    mp::BIOMem bio{pem_cert};
-    auto raw_cert = PEM_read_bio_X509(bio.get(), nullptr, nullptr, nullptr);
-    std::unique_ptr<X509, decltype(X509_free)*> x509{raw_cert, X509_free};
+    QList<QSslCertificate> certs;
+    auto path = QDir(cert_dir).filePath(chain_name);
 
-    if (raw_cert == nullptr)
-        throw std::runtime_error("invalid certificate data");
+    QFile cert_file{path};
+
+    if (cert_file.exists())
+    {
+        // Ensure the file is only read/write by root
+        cert_file.setPermissions(QFile::ReadOwner | QFile::WriteOwner);
+
```

Comment:
I put this here to ensure the permissions are *always* correct regardless of the file operation taking place.  I want to avoid someone mucking with the permissions while Multipass is running and we would never detect and/or set the correct permissions.  If you don't think it's necessary to account for that case, then I'm fine removing this.

Reply:
Yeah if someone is mucking with the permissions they're on their own. We'll overwrite the next time we write it, but I don't think we need it on read.

Reply:
Ack, I'll remove this then.

Reply:
Done.